spec/commands/binstubs_spec.rb

```diff
@@ -124,11 +126,42 @@
         bundle! "binstubs bundler rack prints_loaded_gems"
       end
 
-      let(:system_bundler_version) { Bundler::VERSION }
+      context "when system bundler was used" do
+        # When environment has a same version of bundler as default gems.
+        # `system_gems "bundler-x.y.z"` will detect system binstub.
+        # We need to avoid it by virtual version of bundler.
+        let(:system_bundler_version) { Gem::Version.new(Bundler::VERSION).bump.to_s }
+
+        before do
+          gemfile <<-G
+            source "https://rubygems.org"
+            gem "rack"
+            gem "prints_loaded_gems"
+          G
+
+          lockfile <<-G
+          GEM
+            remote: https://rubygems.org
+            specs:
+              prints_loaded_gems (1.0)
+              rack (1.2)
+
+          PLATFORMS
+            ruby
+
+          DEPENDENCIES
+            prints_loaded_gems
+            rack
+
+          BUNDLED WITH
+             #{system_bundler_version}
+          G
+        end
 
-      it "runs bundler" do
-        sys_exec! "#{bundled_app("bin/bundle")} install"
-        expect(out).to eq %(system bundler #{system_bundler_version}\n["install"])
+        it "runs bundler" do
+          sys_exec! "#{bundled_app("bin/bundle")} install"
+          expect(out).to eq %(system bundler #{system_bundler_version}\n["install"])
+        end
```

Comment:
Actually I don't think _any_ of the above changes are actually needed and we can leave things as they are in master? :thinking: